lib/src/backend/invoker.dart

```diff
@@ -357,7 +371,8 @@ class Invoker {
             // Use the invoker as a key so that multiple invokers can have different
             // outstanding callback counters at once.
             _counterKey: outstandingCallbacksForBody,
-            _closableKey: true
+            _closableKey: true,
+            #runCount: _runCount + 1
```

Comment:
I think it's a little confusing that this gets incremented both here and in `new Future()`. Can we increment it once before we called `runZonedWithValue()`?